### DISCUSSION TITLE
Why middleware cannot redirect to page sign-in

### DISCUSSION THREAD

**Discussion by namnguyen6595** [vercel/next.js#52072] [Help]
### Summary

<img width="669" alt="Screenshot 2023-07-01 at 5 55 53 PM" src="https://github.com/vercel/next.js/assets/40655198/f5dd206e-f0ec-4795-aae9-61262e2f4198">

When i want to redirect to page sign-in, i have error from browser that "ERR_TOO_MANY_REDIRECTS
![Screenshot 2023-07-01 at 5 57 17 PM](https://github.com/vercel/next.js/assets/40655198/ea0652c4-3b28-493f-89ef-3fbbd9e59527)
"

How i can resolve this problem?

**Comment by rahXephonz** [comment]
Is this middleware for

guarding pages with authorizations?
I see on your code the all pages is need authorization right? Please specific a pages that you want to adding guarding pages.

Or please tell me what you want with your custom middleware

**Reply by namnguyen6595** [reply]
i want all pages is need authorization. If token is expired then redirect to page sign-in. Because i don't want to add one by one page

**Comment by rahXephonz** [comment]
> i want all pages is need authorization. If token is expired then redirect to page sign-in. Because i don't want to add one by one page

maybe you can edit of your response
```ts
import { NextRequest, NextResponse } from "next/server";

export async function middleware(request: NextRequest) {
  console.log("request", request.cookies);

  const cookiesExist = request.cookies && request.cookies.get("x-mf-access-token")?.value;

  if (!cookiesExist) {
    const url = new URLSearchParams("/sign-in", request.url);

    return NextResponse.redirect(url);
  }

  return NextResponse.next();
}
```

into this
